### PR TITLE
[FIX] base: res_config_settings allow deferred field


### PR DESCRIPTION
**Pull request by nle-odoo**
[FIX] base: res_config_settings allow deferred field

If a deferred field, for example with auth_user_policy installed:

```html
<field name="f" password="True" widget="password_meter"/>
```

is inside the res_config_settings view, the rendering will be deferred
which is not taken into account when initializing the content
(_initModules) or attaching to DOM (after start).

Thus we would would get an error and have no access to settings.

The issue that happened in 12.0 was solved with 23bec7f and refactored
in between in 5bbbd25, but the heart of the issue was not solved in
the case of a customization.

With this commit, we do not lose the deferred when it is necessary to
keep them in the case of deferred while rendering.

Without fix, the added test fails with a traceback and message:

 TypeError: Cannot read property 'settingView' of undefined

opw-1890343

### CODE REVIEW

File: odoo/addons/base/static/tests/base_settings_tests.js [hunk 1]
```diff
@@ -178,5 +180,82 @@ QUnit.module('base_settings_tests', {
...
+                '</form>',
+        }).then(function (form) {
+            assert.notOk(form.$('.oe_form_configuration').is(':empty'),
+               "render should finish once deferred fields are resolved");
+            assert.ok(form.$('.tab.selected').is('[data-key="crm"]'));
+            assert.ok(form.$('.tab').is('[data-key="otherapp"]'));
+            assert.ok(form.$('.app_settings_block[data-key="crm"]').is(':visible'));
+            assert.ok(form.$('.app_settings_block[data-key="otherapp"]').is(':hidden'));
+            assert.ok($(document.activeElement).is('.searchInput'), "search has focus");
+            form.destroy();
```
Comment (aab-odoo) [inline odoo/addons/base/static/tests/base_settings_tests.js]: i'd do this in always handler as well

File: odoo/addons/base/static/tests/base_settings_tests.js [hunk 2]
```diff
@@ -2,8 +2,10 @@ odoo.define('base.settings_tests', function (require) {
 "use strict";
 
 var testUtils = require('web.test_utils');
+var field_registry = require('web.field_registry');
```
Comment (aab-odoo) [inline odoo/addons/base/static/tests/base_settings_tests.js]: alphab order!!!

File: odoo/addons/base/static/tests/base_settings_tests.js (hunk 1 above)
Comment (nle-odoo) [reply odoo/addons/base/static/tests/base_settings_tests.js]: if we have a `fail` we do not have `form` variable in `createAsyncView`.

so I think either we keep it as is, or put everything in `then` and expect createAsyncView to never fail

Comment (aab-odoo) [reply odoo/addons/base/static/tests/base_settings_tests.js]: haha you're right :-) sorry

**Comment by nle-odoo** [comment]
robodoo r+

**Pull request closed without merging**